source/Projectile.cpp

```diff
@@ -199,6 +198,43 @@ void Projectile::Move(vector<Visual> &visuals, vector<Projectile> &projectiles)
 			}
 		}
 	}
+	// Homing weapons that have lost their lock have a chance to get confused
+	// and turn in a random direction ("go haywire"). Each tracking method has
+	// a different haywire condition. Weapons with multiple tracking methods
+	// only go haywire if all of the tracking methods have gotten confused.
+	else if(target && homing)
+	{
+		bool infraredConfused = true;
+		bool opticalConfused = true;
+		bool radarConfused = true;
+
+		// Infrared: proportional to tracking quality
+		if (weapon->InfraredTracking())
+			infraredConfused = Random::Real() > weapon->InfraredTracking();
+
+		// Optical: proportional tracking quality
+		if (weapon->OpticalTracking())
+			opticalConfused = Random::Real() > weapon->OpticalTracking();
+
+		// Radar: If the target has no jamming, then proportional to tracking
+		// quality. If the target does have jamming, then it's proportional to
+		// tracking quality, the strength of target's jamming, and the distance
+		// to the target (jamming power attenuates with distance).
+		if (weapon->RadarTracking())
+		{
+			if (target->Attributes().Get("radar jamming") == 0)
+				radarConfused = Random::Real() > weapon->RadarTracking();
+			else
+			{
+				// Closeness to target; 0 when just fired, 1 when impacting
+				double closenessTotarget = 1 + weapon->RadarTracking() - (1 / (weapon->Range() / position.Distance(target->Position())));
```

Comment:
Risk of division by zero if position and target->Position somehow become the same. (Unlikely, but possible when ships have open space in the middle.)

Reply:
Are you sure that RadarTracking needs to be added to closenessToTarget? (And should closenessToTarget have 2 capital Ts?)

Reply:
> Risk of division by zero if position and target->Position somehow become the same

Fixed.

> Are you sure that RadarTracking needs to be added to closenessToTarget

Yes, because I want the closeness factor to be affected by the quality of radar tracking.

> (And should closenessToTarget have 2 capital Ts?)

Style thing, but it makes sense to me. :)